modules/index/src/test/java/org/apache/ignite/internal/index/IndexBuilderTest.java

```diff
@@ -149,6 +154,16 @@ void testIndexBuildWithReplicationTimeoutException() {
         verify(replicaService, times(2)).invoke(any(ClusterNode.class), any(BuildIndexReplicaRequest.class));
     }
 
+    @Test
+    void testScheduleBuildIndexAfterDisasterRecovery() {
+        CompletableFuture<Void> listenCompletionIndexBuildingAfterDisasterRecoveryFuture =
+                listenCompletionIndexBuildingAfterDisasterRecovery(INDEX_ID, TABLE_ID, PARTITION_ID);
+
+        scheduleBuildIndexAfterDisasterRecovery(INDEX_ID, TABLE_ID, PARTITION_ID, List.of(rowId(PARTITION_ID)));
+
+        assertThat(listenCompletionIndexBuildingAfterDisasterRecoveryFuture, willCompleteSuccessfully());
```

Comment:
Could we also assert that onBuildCompletion method in listener is not triggered? The same for testScheduleBuildIndex, that afterRecovery is not triggered